academy/agent.py

```diff
@@ -107,14 +107,14 @@ def __init__(
         self,
         behavior: BehaviorT,
         *,
-        agent_id: AgentId[BehaviorT],
+        agent_info: RegistrationInfo[BehaviorT],
         exchange: ExchangeFactory,
         config: AgentRunConfig | None = None,
     ) -> None:
-        self.agent_id = agent_id
+        self.agent_info = agent_info
```

Comment:
It might be worth keeping an `agent_id` attribute as a property since it is accessed so many times (would also reduce the loc changed).

```python
@property
def agent_id(self) -> AgentId[BehaviorT]:
    return self.agent_info.agent_id
```